planning/behavior_velocity_planner/src/scene_module/bus_stop/scene.cpp

```diff
@@ -0,0 +1,619 @@
+    debug_data_->predicted_vel_kmph = velocity_buffer_lpf_.back() * 3.6;
+    debug_data_->pushPredictedVelKmph(velocity_buffer_.back() * 3.6);
+    debug_data_->pushPredictedVelLpfKmph(velocity_buffer_lpf_.back() * 3.6);
+    debug_data_->publishDebugValue();
+  }
+  debug_data_->nearest_point = nearest_point_with_dist.first;
+  debug_data_->is_safe_velocity = is_safe_velocity;
+  debug_data_->is_obstacle_on_the_side = is_obstacle_on_the_side;
+  if (is_obstacle_on_the_side) {
+    RCLCPP_DEBUG_STREAM(logger_, "obstacle is on the side of the vehicle");
+  }
+
+  // calculate stop point for the stop line
+  const auto path_idx_with_pose = calcStopPoint(*path);
+  if (!path_idx_with_pose) {
+    RCLCPP_WARN_STREAM(logger_, "failed to calculate stop point.");
+    return true;
+  }
+  const auto stop_dist = calcStopDistance(*path, *path_idx_with_pose);
+  const auto & stop_pose = path_idx_with_pose->second;
+  const auto & base_link_pose = planner_data_->current_pose.pose;
+
+  if (current_state == State::STOP) {
+    // if RTC is activated, do not insert zero velocity and go
+    if (isRTCActivated(stop_dist, false)) {
+      return true;
+    }
+
+    planning_utils::insertStopPoint(stop_pose.position, *path);
+
+    // create virtual wall at the head of the ego vehicle
+    debug_data_->stop_poses.push_back(base_link_pose);
+
+    return true;
+  }
+
+  if (current_state == State::READY) {
+    // if RTC is activated, do not insert zero velocity and go
+    if (isRTCActivated(stop_dist, false)) {
+      return true;
+    }
+
+    planning_utils::insertStopPoint(stop_pose.position, *path);
+
+    // create virtual wall at the head of the ego vehicle
+    debug_data_->stop_poses.push_back(base_link_pose);
+
+    // Set Turn Indicator
+    turn_indicator_->setTurnSignal(TurnIndicatorsCommand::ENABLE_RIGHT, clock_->now());
+    turn_indicator_->publish();
+
+    return true;
+  }
+
+  if (current_state == State::GO) {
+    // if RTC is activated, do not insert zero velocity and go
+    if (isRTCActivated(stop_dist, true)) {
+      return true;
+    }
+
+    planning_utils::insertStopPoint(stop_pose.position, *path);
+
+    // create virtual wall at the head of the ego vehicle
+    debug_data_->stop_poses.push_back(base_link_pose);
+
+    return true;
+  }
+
+  return true;
+}
+
+LineString2d BusStopModule::getStopLineGeometry2d() const
+{
+  const lanelet::ConstLineString3d stop_line = bus_stop_reg_elem_.stopLine();
+  return planning_utils::extendLine(
+    stop_line[0], stop_line[1], planner_data_->stop_line_extend_length);
+}
+
+boost::optional<PathIndexWithPose> BusStopModule::calcStopPoint(const PathWithLaneId & path) const
+{
+  // Get stop line geometry
+  const auto stop_line = getStopLineGeometry2d();
+
+  // Get stop point
+  // In this module, the ego vehicle want to keep stopping at the first stop position
+  // set the stop_margin so that the ego vehicle don't approach the stop line
+  const double stop_margin = 3.0;
```

Comment:
Does this need to be param?

Reply:
This value isn't meant to be tuned, but I think it is better to be this value as a parameter and add explanation to the param file to make it clearer.
So I fixed.
https://github.com/tier4/autoware.universe/pull/339/commits/31d8b8746d6b942901f7747d43837fba563998ce